Rubberduck.VBEEditor/SafeComWrappers/Abstract/HostApplicationBase.cs

```diff
@@ -149,8 +149,9 @@ public virtual IEnumerable<HostDocument> GetDocuments()
             return result;
         }
 
-        public virtual HostDocument GetDocument(QualifiedModuleName moduleName)
+        public virtual bool TryGetDocument(QualifiedModuleName moduleName, out HostDocument document)
```

Comment:
I'm not sure I'm sold on the `Try` pattern for reference types - it's more appropriate for intrinsics and structs than it is for objects.  The problem is that it makes it difficult to use pattern matching on the return value, i.e.:

`if (foo.GetDocument(bar) is HostDocument host && host.SomeProperty)`

TBH, I'd still null check the `out` parameter even if it returned `true` because it's trivial with more recent c# syntax.

Reply:
`HostDocument` is actually meant to be more like a `readonly struct` but I ended up making it an object -- maybe it was because of the (currently unused) function or some restrictions I ran into using the `readonly struct`. You can think of `HostDocument` as something akin to `QMN`s for host documents --- we simply cache the metadata about the document without giving away ~the keys to the floodgate~ COM accesses.

Reply:
Can it be a struct, or is that a PITA? The main thing here is that the pattern doesn't match the types involved.

Reply:
It did start our as a `readonly struct` originally but for reasons I should have written down, it did not work out. I will give it another try because it really really wants to be a `readonly struct`.